mmv1/templates/terraform/examples/bigquery_dataset_access_authorized_routine.tf.erb

```diff
@@ -0,0 +1,36 @@
+resource "google_bigquery_dataset" "public" {
+  dataset_id  = "<%= ctx[:vars]['public'] %>"
+  description = "This dataset is public"
+}
+
+resource "google_bigquery_routine" "public" {
+  dataset_id      = google_bigquery_dataset.public.dataset_id
+  routine_id      = "sample_routine"
```

Comment:
Is this automatically deleted if the dataset is deleted? And/or is there any issue with quotas for this resource, or name conflicts if parallel test runs happen at the same time? Otherwise it might be worth making this use a var as well.